.github/workflows/tests.yml

```diff
@@ -445,6 +445,168 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui-e2e-tests:
+    name: UI e2e tests
+    runs-on: ubuntu-latest
+    needs: [code-tests, documentation]
+    if: ${{ github.repository == 'canonical/lxd' && github.event_name == 'push' }}
+    env:
+      LXD_DIR: "/var/lib/lxd"
+      LXD_OIDC_CLIENT_ID: ${{ secrets.LXD_UI_OIDC_CLIENT_ID }}
+      LXD_OIDC_ISSUER: ${{ secrets.LXD_UI_OIDC_ISSUER }}
+      LXD_OIDC_AUDIENCE: ${{ secrets.LXD_UI_OIDC_AUDIENCE }}
+      LXD_OIDC_USER: ${{ secrets.LXD_UI_OIDC_TEST_USER }}
+      LXD_OIDC_PASSWORD: ${{ secrets.LXD_UI_OIDC_TEST_PASSWORD }}
+      LXD_OIDC_GROUPS_CLAIM: "lxd-idp-groups"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Tune disk performance
+        uses: ./.github/actions/tune-disk-performance
+
+      - name: Install Go
+        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Install runtime dependencies
+        uses: ./.github/actions/install-lxd-runtimedeps
+
+      - name: Download minio/mc to add to system test dependencies
+        uses: ./.github/actions/download-minio
+
+      - name: Download system test dependencies
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: system-test-deps
+          merge-multiple: true
+          path: /home/runner/go/bin
+
+      - name: Set exec perms on LXD binaries
+        run: |
+          set -eux
+          ls -lR /home/runner/go/bin/
+          chmod uog+x /home/runner/go/bin/*
+
+      - name: Download docs (for the objects.inv.txt file to be available for the e2e tests)
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: documentation
+          merge-multiple: true
+          path: doc/_build
+
+      - name: Make GOCOVERDIR
+        run: |
+          set -eux
+          mkdir -p "${GOCOVERDIR}"
+        if: env.GOCOVERDIR != ''
+
+      - name: Run LXD daemon
+        run: |
+          set -eux
+          echo "root:1000000:1000000000" | sudo tee /etc/subuid /etc/subgid
+          export CGO_CFLAGS="-I/home/runner/work/lxd/lxd-test/vendor/dqlite/include/"
+          export CGO_LDFLAGS="-L/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
```

Comment:
This path seems wrong as the system-test-deps are all unpacked into `~/go/bin`. The `LD_LIBRARY_PATH` suggested earlier should work.

Reply:
Yes, thanks for flagging this :)